mpsl/include/mpsl_fem_protocol_api.h

```diff
@@ -0,0 +1,307 @@
+ *
+ * @param[in] p_activate_event   Pointer to the activation event structure.
+ * @param[in] p_deactivate_event Pointer to the deactivation event structure.
+ *
+ * @pre To activate PA, the corresponding configuration setter of the selected Front End Module must have been called first.
+ *
+ * @note If a timer event is provided, the caller of this function is responsible for starting the timer and configuring its shorts.
+ *       Moreover, the caller is responsible for stopping the timer no earlier than the compare channel of the lowest ID among the provided ones does expire.
+ *
+ * @retval   0             PA activation setup is successful.
+ * @retval   -NRF_EPERM    PA is currently disabled.
+ * @retval   -NRF_EINVAL   PA activation setup could not be performed due to invalid or missing configuration parameters
+ *                         in p_activate_event or p_deactivate_event, or both.
+ */
+int32_t mpsl_fem_pa_configuration_set(const mpsl_fem_event_t * const p_activate_event,
+                                      const mpsl_fem_event_t * const p_deactivate_event);
+
+/**
+ * @brief Clears up the configuration provided by the @ref mpsl_fem_pa_configuration_set function.
+ *
+ * @retval   0             PA activation setup purge is successful.
+ * @retval   -NRF_EPERM    PA is currently disabled.
+ */
+int32_t mpsl_fem_pa_configuration_clear(void);
+
+/**
+ * @brief Sets up LNA using the provided event for the upcoming radio reception.
+ *
+ * Multiple configurations can be provided by repeating calls to this function (that is, you can set the activate and the deactivate event in multiple calls,
+ * and the configuration is preserved between calls).
+ *
+ * The order of calls of this function and its `pa` counterpart must match the order of radio operations.
+ * I.e. if you want to listen first and then send the frame, you need first to issue @ref mpsl_fem_lna_configuration_set and only after @ref mpsl_fem_pa_configuration_set.
+ *
+ * If a @ref mpsl_fem_event_type_t::MPSL_FEM_EVENT_TYPE_TIMER timer event is provided, the LNA will be configured to activate or deactivate at the application-configured time gap
+ * before the timer instance reaches the given register_value. The time gap is set via the corresponding configuration setter of the selected Front End Module.
+ *
+ * If a @ref mpsl_fem_event_type_t::MPSL_FEM_EVENT_TYPE_GENERIC event is provided, the LNA will be configured to activate or deactivate when an event occurs.
+ *
+ * The function sets up the PPIs and the GPIOTE channel to activate LNA for the upcoming radio transmission.
+ * The LNA pin will be active until deactivated, which can happen either by encountering a configured deactivation event or by using @ref mpsl_fem_deactivate_now.
+ *
+ * @param[in] p_activate_event   Pointer to the activation event structure.
+ * @param[in] p_deactivate_event Pointer to the deactivation event structure.

```

Comment:
Is it possible to pass ``NULL`` pointer here to indicate to the library that the user wants to drive PA / LNA deactivation using ``mpsl_fem_deactivate_now`` API?